src/libcore/mem.rs

```diff
@@ -611,3 +611,38 @@ pub const POST_DROP_USIZE: usize = POST_DROP_U64 as usize;
 pub unsafe fn transmute_copy<T, U>(src: &T) -> U {
     ptr::read(src as *const T as *const U)
 }
+
+/// Returns the value of the discriminant for the enum variant in `v`.
+///
+/// The returned discriminant is cast to `u64`, no matter what the actual
```

Comment:
Seems problematic with `#[repr(i/u128)]` potentially happening once we get `i/u128` primitives. Are we gonna forbid such attribute?